pkgs/by-name/on/ondsel/package.nix

```
@@ -0,0 +1,36 @@
+{ appimageTools, lib, fetchurl }:
+
+let
+  pname = "ondsel";
+  version = "2024.1.0";
+  patch = "35694";
+
+  src = fetchurl {
+    # TODO: integrate version rest...
+    url = "https://github.com/Ondsel-Development/FreeCAD/releases/download/${version}/Ondsel_ES_${version}.${patch}-Linux-x86_64.AppImage";
+    hash = "sha256-UNltd0fF2MvVbmANcEaqgEBsrerxqnGCHZXMLGp6OKw=";
+  };
+in
+appimageTools.wrapType2 rec {
+  inherit pname version src;
+
+  extraInstallCommands =
+    let contents = appimageTools.extractType2 { inherit pname version src; };
+    in ''
+      mv $out/bin/${pname}-${version} $out/bin/${pname}
+
+      install -m 444 -D ${contents}/com.ondsel.ES.desktop $out/share/applications/${pname}.desktop
+
+      substituteInPlace $out/share/applications/${pname}.desktop \
+        --replace 'Exec=AppRun' 'Exec=${pname}'
+      cp -r ${contents}/usr/share/icons $out/share
+    '';
+
+
+  meta = with lib; {
+    description = "An engineering suite with an open-source heart (Solo version)";
+    homepage = "https://www.ondsel.com";
+    license = licenses.lgpl2Plus;
+    platforms = [ "x86_64-linux" ];
```

I suppose these are supported as well, even though they haven't been packaged. Feel free to mark darwin as broken if ofborg complains and you don't want to deal with it.

```suggestion
    platforms = [ "x86_64-linux" "x86_64-linux" "aarch64-darwin" ];
```

---

Don't you mean ARM Linux with the second "x86_64-linux" platform?

I could also package it for ARM Linux, but probably not (yet) for MacOS, but I don't know how. Is it something like

```nix
src = if stdenv.isX86() fetchUrl { 
} else if ... fetchUrl {
};
```?

---

Nevermind, I found another package that packages for different platforms. I've prepared the package for other platforms, but marked all others (ARM Linux, ARM + x86_64 Darwin) as broken.